src/App.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useState, useCallback, useRef } from "react";
+import Header from "./components/Header";
+import Home from "./components/Home";
+import { SecretDataType } from "./secretDataType";
+
+function App() {
+  const secretText = "inject"; // Secret Key
+  const keyTime = 5000; //Typing duration
+  const showDataTime = 15000; //ShowData
+
+  const intervalRef: { current: NodeJS.Timeout | null } = useRef(null);
```

Comment:
I think it's okay with just const intervalRef = React.useRef<number | null>(null);